addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/IPBridgeHandler.java

```diff
@@ -246,15 +248,24 @@ private synchronized void reconnect() {
 
     private boolean login(IPBridgeConfig config) throws IOException, InterruptedException {
         this.session.open(config.getIpAddress());
-
         this.session.waitFor("login:");
-        this.session.writeLine(config.getUser() != null ? config.getUser() : DEFAULT_USER);
-        this.session.waitFor("password:");
-        this.session.writeLine(config.getPassword() != null ? config.getPassword() : DEFAULT_PASSWORD);
 
-        MatchResult matchResult = this.session.waitFor("(login:|GNET>)");
+        // Sometimes the Lutron Smart Bridge Pro will request login more than once.
+        int attemptNum = 1;
+        while (attemptNum++ < MAX_LOGIN_ATTEMPTS) {
```

Comment:
I believe you are correct, it will not. It will likely be much more clear if it were a for loop.